src/vs/base/browser/ui/resourceviewer/resourceViewer.ts

```diff
@@ -157,9 +158,21 @@ export class ResourceViewer {
 		else {
 			$(container)
 				.empty()
-				.span({
+				.append($('p', {
 					text: nls.localize('nativeBinaryError', "The file will not be displayed in the editor because it is either binary, very large or uses an unsupported text encoding.")
-				});
+				}))
+				.append($('a', {
+					role: 'button',
+					class: 'open-external',
+					text: nls.localize('resourceOpenExternal', "Try opening file in external program?")
+				}).on(DOM.EventType.CLICK, (e) => {
+					windowsService.openExternal(descriptor.resource.toString()).then(didOpen => {
```

Comment:
Do we have an understanding what this native call does when the file is an executable? Also, how does this behave when you are seeing a binary file which is associated with VS Code, you click it only to see the same issue again because it opens in VS Code?